python/tskit/trees.py

```diff
@@ -3713,22 +3719,37 @@ def haplotypes(self, impute_missing_data=False, missing_data_character="-"):
 
         :return: An iterator over the haplotype strings for the samples in
             this tree sequence.
-        :param bool impute_missing_data: If True, the allele assigned to any
-            isolated samples is the ancestral state; that is, we impute
-            missing data as the ancestral state. Default: False.
+        :param bool isolated_as_missing: If True, the allele assigned to
+            missing (isolated) samples is ``missing_data_character``. If False,
+            missing (isolated) samples will be assigned the ancestral state or
+            that of the mutation above them, if any.
+            Default: True.
         :param str missing_data_character: A single ascii character that will
             be used to represent missing data.
             If any normal allele contains this character, an error is raised.
             Default: '-'.
+        :param bool impute_missing_data:
+            *Deprecated in 0.3.0. Use ``isolated_as_missing``, but inverting value.
+            Will be removed in a future version*
         :rtype: collections.abc.Iterable
         :raises: TypeError if the ``missing_data_character`` or any of the alleles
             at a site or the are not a single ascii character.
         :raises: ValueError
             if the ``missing_data_character`` exists in one of the alleles
         """
+        if impute_missing_data is not None:
+            warnings.warn(
+                "The impute_missing_data parameter was deprecated in 0.3.0 and will"
+                " be removed. Use ``isolated_as_missing``, but inverting the value.",
```

Comment:
perhaps "Use ``isolated_as_missing=False`` instead of ``impute_missing_data=True``."?

Reply:
since "invert" the value might be confusing (which value?)

Reply:
Done